src/init.cpp

```diff
@@ -868,6 +878,55 @@ void ThreadImport(std::vector<fs::path> vImportFiles)
     }
 }
 
+void PeriodicStats()
+{
+    CCoinsStats stats;
+    FlushStateToDisk();
+    if (GetUTXOStats(pcoinsdbview.get(), stats)) {
+        statsClient.gauge("utxoset.tx", stats.nTransactions, 1.0f);
+        statsClient.gauge("utxoset.txOutputs", stats.nTransactionOutputs, 1.0f);
+        statsClient.gauge("utxoset.dbSizeBytes", stats.nDiskSize, 1.0f);
+        statsClient.gauge("utxoset.blockHeight", stats.nHeight, 1.0f);
+        statsClient.gauge("utxoset.totalAmount", (double)stats.nTotalAmount / (double)COIN, 1.0f);
+    } else {
+        // smth went wrong
+        LogPrintf("%s: GetUTXOStats failed\n", __func__);
+    }
+    statsClient.gauge("transactions.txCacheSize", pcoinsTip->GetCacheSize(), 1.0f);
+
+    // short version of GetNetworkHashPS(120, -1);
+    CBlockIndex *tip;
+    {
+        LOCK(cs_main);
+        tip = chainActive.Tip();
+        assert(tip);
+    }
+    CBlockIndex *pindex = tip;
+    int64_t minTime = pindex->GetBlockTime();
+    int64_t maxTime = minTime;
+    for (int i = 0; i < 120 && pindex->pprev != nullptr; i++) {
+        pindex = pindex->pprev;
+        int64_t time = pindex->GetBlockTime();
+        minTime = std::min(time, minTime);
+        maxTime = std::max(time, maxTime);
+    }
+    arith_uint256 workDiff = tip->nChainWork - pindex->nChainWork;
+    int64_t timeDiff = maxTime - minTime;
+    double nNetworkHashPS = workDiff.getdouble() / timeDiff;
+
+    statsClient.gauge("network.hashesPerSecond", nNetworkHashPS);
+    statsClient.gauge("network.terahashesPerSecond", nNetworkHashPS / 1e12);
+    statsClient.gauge("network.petahashesPerSecond", nNetworkHashPS / 1e15);
+    statsClient.gauge("network.exahashesPerSecond", nNetworkHashPS / 1e18);
+    // No need for cs_main, we never use null tip here
+    statsClient.gauge("network.difficulty", (double)GetDifficulty(tip));
+
+    statsClient.gauge("transactions.mempool.totalTransactions", mempool.size(), 1.0f);
+    statsClient.gauge("transactions.mempool.totalTxBytes", (int64_t) mempool.GetTotalTxSize(), 1.0f);
+    statsClient.gauge("transactions.mempool.memoryUsageBytes", (int64_t) mempool.DynamicMemoryUsage(), 1.0f);
+    statsClient.gauge("transactions.mempool.minFeePerKb", mempool.GetMinFee(gArgs.GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000).GetFeePerK(), 1.0f);
```

Comment:
Why is this 1000000?

Reply:
Because `DEFAULT_MAX_MEMPOOL_SIZE` is in Mb and `CTxMemPool::GetMinFee(size_t sizelimit)` expects `sizelimit` to be in bytes.